snap/info_snap_yaml.go

```diff
@@ -303,6 +315,67 @@ func infoSkeletonFromSnapYaml(y snapYaml) *Info {
 	return snap
 }
 
+func setComponentsFromSnapYaml(y snapYaml, snap *Info, strk *scopedTracker) {
```

Comment:
Maybe parts of this should be shared with `setHooksFromSnapYaml`

Reply:
I did a little refactoring to address this in 904122dcb2b5493b345c234fba928116a0b5ab76.